Gemfile

```diff
@@ -36,6 +36,8 @@ group :system_tests do
   gem "serverspec",                                   require: false, platforms: [:ruby]
 end
 
+gem "scanf", require: false if Gem::Version.new(RUBY_VERSION.dup) >= Gem::Version.new('2.7.0')
+
```

Comment:
I'm thinking this is going to affect modules other than puppetdb, and would be better to add this to https://github.com/puppetlabs/puppet-module-gems/blob/117040802f7981aa097b16c03ffbc455d1f83e8f/config/dependencies.yml#L143-L155. I'll ping the modules folks

Reply:
I think this is actually a puppet bug. The puppet gem should express a runtime gem dependency on `scanf`. Filed as https://tickets.puppetlabs.com/browse/PUP-10797

Reply:
This will be released in puppet 7.1.0